qiskit/providers/aer/backends/aer_compiler.py

```diff
@@ -0,0 +1,218 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2018, 2019, 2021
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Compier to convert Qiskit control-flow to Aer backend.
+"""
+from qiskit.circuit import Instruction, QuantumCircuit
+from qiskit.circuit.controlflow import (
+    WhileLoopOp,
+    ForLoopOp,
+    IfElseOp,
+    BreakLoopOp,
+    ContinueLoopOp)
+
+
+class AerMark(Instruction):
+    """
+    Mark instruction
+
+    This instruction is a destination of jump instruction.
+    Conditional is not allowed in Aer controller.
+    """
+
+    def __init__(self, name, num_qubits):
+        super().__init__("mark", num_qubits, 0, [name])
+
+
+class AerJump(Instruction):
+    """
+    Jump instruction
+
+    This instruction sets a program counter to specified mark instruction.
+    """
+
+    def __init__(self, jump_to, num_qubits):
+        super().__init__("jump", num_qubits, 0, [jump_to])
+
+
+class AerCompiler:
```

Comment:
It would be good if this could be built into an aer `assemble` method so it only happens when we construct the qobj payload for the C++ simulator (same as with the noise replacement in #1350) so we can avoid iterating over all the circuits too many times.

Reply:
I integrated `AerCompiler` into `AerBackend`.